docs/cicd/github-cd-template.yml

```diff
@@ -0,0 +1,97 @@
+# This is just an example workflow for continuous deployment.
+# You should customize it to meet your own requirements.
+name: 'Continuous Deployment'
+on:
+  # When new commits are pushed onto the main branch.
+  push:
+    branches:
+      - main
+jobs:
+  buildAndPublish:
+    runs-on: ubuntu-latest
+    # You can uncomment the line below to use environments (refer to https://docs.github.com/en/actions/reference/environments). 
+    #environment: test_environment
+    env:
+      AZURE_ACCOUNT_NAME: ${{secrets.AZURE_ACCOUNT_NAME}}
+      AZURE_ACCOUNT_PASSWORD: ${{secrets.AZURE_ACCOUNT_PASSWORD}}
+      AZURE_SUBSCRIPTION_ID: ${{secrets.AZURE_SUBSCRIPTION_ID}}
+      AZURE_TENANT_ID: ${{secrets.AZURE_TENANT_ID}}
+      M365_ACCOUNT_NAME: ${{secrets.M365_ACCOUNT_NAME}}
+      M365_ACCOUNT_PASSWORD: ${{secrets.M365_ACCOUNT_PASSWORD}}
+
+    steps:
+      # Setup environment.
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '14'
+          
+      - name: Checkout the code
+        uses: actions/checkout@v2
+        
+      # Build the project.
+      # The way to build the current project depends on how you scaffold it.
+      # Different folder structures require different commands set.
+      # 'npm ci' is used here to install dependencies and it depends on package-lock.json.
+      # If you prefer to use 'npm ci', please make sure to commit package-lock.json first, or just change it to 'npm install'.
+      - name: Build the project
+        run: cd tabs && npm ci && npm run build
+
+      # Run unit test.
+      # Currently, no opinioned solution for unit test provided during scaffolding, so,
+      # set up any unit test framework you prefer (for example, mocha or jest) and update the commands accordingly in below.
+      - name: Run Unit Test
+        run: cd tabs && npm run test
+
+      # We suggest to do the provision steps by case manually or in a separated workflow, so just comment the following steps for references.
+      # After provision, you should commit .fx/env.default.json into the repository for later use.
+      # You should pick required secrets from .fx/default.userdata, and save them in the repository secrets (https://docs.github.com/en/actions/reference/encrypted-secrets) which can be refered by the step with name 'Generate default.userdata'. 
```

Comment:
This means out of the box the action is not going to work right? That the user needs to do these things in order for deploy to work without provision?

Reply:
Yes, that's by design. Since provision is an operation which will be performed by case, we expect users to do it separately.

Reply:
Can we add that to the README? Like a section that is REQUIREMENTS or something, so that people can see. Otherwise the only way I would find out is to read the YML file.

Reply:
There exists one `Note:` section under `Customize CD Workflow` inside README to describe this. How about it?